### PR TITLE
refactor: add docs + cleanup MA and MABase with small opt

### PR DESCRIPTION
**Pull request by fangting-alchemy**
* Add NatSpecs
* Small refactor on comments and code
* tiny opt by rid of one error

### CODE REVIEW

File: src/account/ModularAccountBase.sol [hunk 1]
```diff
@@ -100,7 +104,8 @@ abstract contract ModularAccountBase is
     error ValidationFunctionMissing(bytes4 selector);
     error DeferredActionNonceInvalid();
     error DeferredActionSignatureInvalid();
-    error CreateFailed();
+    // This error is trigged in performCreate and performCreate2
+    // error CreateFailed();
```
Comment (adamegyed) [inline src/account/ModularAccountBase.sol]: This doesn't affect codesize - we should keep it in for the ABI to contain it.

File: src/account/ModularAccountBase.sol [hunk 2]
```diff
@@ -43,6 +43,10 @@ import {ModularAccountView} from "./ModularAccountView.sol";
 import {ModuleManagerInternals} from "./ModuleManagerInternals.sol";
 import {TokenReceiver} from "./TokenReceiver.sol";
 
+/// @title Modular Account
+/// @author Alchemy
+/// @notice This abstract account is a modular account that is compliant with ERC-6900 standard. It supports
+/// deferred account creation.
```
Comment (adamegyed) [inline src/account/ModularAccountBase.sol]:
```suggestion
/// @title Modular Account Base
/// @author Alchemy
/// @notice This abstract account is a modular account that is compliant with ERC-6900 standard. It supports
/// deferred actions during validation.
```

File: src/account/ModularAccount.sol [hunk 3]
```diff
@@ -6,6 +6,11 @@ import {IEntryPoint} from "@eth-infinitism/account-abstraction/interfaces/IEntry
 
 import {ModularAccountBase} from "./ModularAccountBase.sol";
 
+/// @title Modular Account
+/// @author Alchemy
+/// @notice This contract allow initionalizing with a passed in validation config (of a validation module), which
+/// will be installed onto the account. Therefore, the account can be used with the validation after or with (if
+/// deferred) creation.
```
Comment (adamegyed) [inline src/account/ModularAccount.sol]:
```suggestion
/// @notice This contract allow initializing with a validation config to install on the account.
```

Not sure what the latter sentence is saying - could you clarify?

Comment (fangting-alchemy) [reply src/account/ModularAccount.sol]: Not necessary anymore. Updated

File: src/account/ModularAccount.sol [hunk 4]
```diff
@@ -6,6 +6,10 @@ import {IEntryPoint} from "@eth-infinitism/account-abstraction/interfaces/IEntry
 
 import {ModularAccountBase} from "./ModularAccountBase.sol";
 
+/// @title Modular Account
+/// @author Alchemy
+/// @notice This contract allow initionalizing with a validation config (of a validation module) to be installed on
```
Comment (Zer0dot) [inline src/account/ModularAccount.sol]:
```suggestion
/// @notice This contract allow initializing with a validation config (of a validation module) to be installed on
```

File: src/account/ModularAccountBase.sol [hunk 5]
```diff
@@ -43,6 +43,10 @@ import {ModularAccountView} from "./ModularAccountView.sol";
 import {ModuleManagerInternals} from "./ModuleManagerInternals.sol";
 import {TokenReceiver} from "./TokenReceiver.sol";
 
+/// @title Modular Account Base
+/// @author Alchemy
+/// @notice This abstract account is a modular account that is compliant with ERC-6900 standard. It supports
```
Comment (Zer0dot) [inline src/account/ModularAccountBase.sol]:
```suggestion
/// @notice This abstract contract is a modular account that is compliant with ERC-6900 standard. It supports
```

**Review by adamegyed — APPROVED** [review]

**Pull request merged**